### ISSUE TITLE
Enhance learning rate scaling and adjustment beyond linear scaling

### ISSUE THREAD

**Issue by tgaddair** [horovod/horovod#2135]
Breaking the frontiers - examples/results with extremely large-scale training.

- Custom per-layer learning rates from [32k batch size paper](https://arxiv.org/pdf/1708.03888.pdf)
- Zig-zag and other LR adjustment techniques - [here](https://medium.com/@bushaev/improving-the-way-we-work-with-learning-rate-5e99554f163b).
- Multi-step SGD (don't increase LR too much, make few steps based on the same preprocessed and loaded input data) using duplicating generator.
